src/commonMain/kotlin/fr/acinq/lightning/wire/LightningMessages.kt

```diff
@@ -1595,112 +1638,191 @@ data class OnionMessage(
 }
 
 /**
- * When we don't have enough incoming liquidity to receive a payment, our peer may open a channel to us on-the-fly to carry that payment.
- * This message contains details that allow us to recalculate the fee that our peer will take in exchange for the new channel.
- * This allows us to combine multiple requests for the same payment and figure out the final fee that will be applied.
- *
- * @param chainHash chain we're on.
- * @param fundingSatoshis total capacity of the channel our peer will open to us (some of the funds may be on their side).
- * @param amountMsat payment amount covered by this new channel: we will receive push_msat = amountMsat - fees.
- * @param payToOpenMinAmountMsat minimum amount for a pay-to-open to be attempted, this should be compared to the total amount in the case of an MPP payment.
- * @param payToOpenFeeSatoshis fees that will be deducted from the amount pushed to us (this fee covers the on-chain fees our peer will pay to open the channel).
- * @param paymentHash payment hash.
- * @param expireAt after the proposal expires, our peer will fail the payment and won't open a channel to us.
- * @param finalPacket onion packet that we would have received if there had been a channel to forward the payment to.
+ * This message is sent when an HTLC couldn't be relayed to our node because we don't have enough inbound liquidity.
+ * This allows us to treat it as an incoming payment, and request on-the-fly liquidity accordingly if we wish to receive that payment.
+ * If we accept the payment, we will send an [OpenDualFundedChannel] or [SpliceInit] message containing [ChannelTlv.RequestFundingTlv].
+ * Our peer will then provide the requested funding liquidity and will relay the corresponding HTLC(s) afterwards.
  */
-data class PayToOpenRequest(
+data class WillAddHtlc(
```

Comment:
From a spec point of view, it would be hacky to have a dummy `id` field. Also, what would you do with the tlv stream of the `update_add_htlc`? Just use an empty one/ignore it?

I'm also afraid it would add more confusion and we'd risk incorrectly confusing an `UpdateAddHtlc` actually coming from `WillAddHtlc` and a normal one, and creating bugs because of it...